src/parser/binary_parser.ml

```diff
@@ -1020,17 +1015,19 @@ let sections_iterate (modul : Binary.modul) (input : Input.t) =
   let func =
     let values =
       let id = ref (-1) in
-      List.map2
-        (fun typeidx (locals, body) ->
-          incr id;
-          Indexed.return !id
-            (Runtime.Local
-               { type_f = Array.get block_type_array typeidx
-               ; locals
-               ; body
-               ; id = None
-               } ) )
-        typeidx_list_func locals_code_list
+      try
+        List.map2
+          (fun typeidx (locals, body) ->
+            incr id;
+            Indexed.return !id
+              (Runtime.Local
+                 { type_f = Array.get block_type_array typeidx
+                 ; locals
+                 ; body
+                 ; id = None
+                 } ) )
+          typeidx_list_func locals_code_list
+      with Invalid_argument _ -> assert false
```

Comment:
I'm not sure what you're trying to do here but you should not catch `Invalid_argument`. If the length of the two lists may differ, it's better to test it explicitly before calling `List.map2`.

Reply:
This specific test existed ;)